node_modules/col-activities/lib/notifications/weekly.js

```diff
@@ -184,24 +182,37 @@ var getCourseData = function(course, callback) {
     }
   };
 
-  // 1. Get all active users in the course.
+  // 1. Get all active users in the course and their total points.
   var options = {
     'enrollmentStates': CollabosphereConstants.ENROLLMENT_STATE.ACTIVE,
     'includeEmail': true
   };
-  UsersAPI.getAllUsers(ctx, options, function(err, users) {
+  UsersAPI.getLeaderboard(ctx, options, function(err, users) {
     if (err) {
       log.error({'err': err, 'course': ctx.course.id}, 'Failed to retrieve the users for a course');
       return callback(err);
     }
 
-    // Index users by id for quick lookup.
-    users = _.chain(users)
-      .map(function(user) {
-        return user.toJSON();
-      })
-      .indexBy('id')
-      .value();
+    // Calculate user ranks. Users are returned in descending order of score, so rank can be derived from array index. Rank 
+    // is defined as the number of users with a higher score, plus one; e.g., scores [200, 100, 50, 50, 25] map to ranks
+    // [1, 2, 3, 3, 5].
+    // TODO: Store this week's ranks and retrieve last week's ranks.
```

Comment:
If we will be storing this week's and last week's rank, is it necessary to calculate the rank here?